ibc-core/ics04-channel/src/handler/recv_packet.rs

```diff
@@ -248,18 +248,11 @@ where
             }
         }
         Order::Unordered => {
-            let receipt_path_on_b = ReceiptPath::new(
-                &msg.packet.port_id_on_a,
-                &msg.packet.chan_id_on_a,
-                msg.packet.seq_on_a,
-            );
-            let packet_rec = ctx_b.get_packet_receipt(&receipt_path_on_b);
-            match packet_rec {
-                Ok(_receipt) => {}
-                Err(ContextError::PacketError(PacketError::PacketReceiptNotFound { sequence }))
-                    if sequence == msg.packet.seq_on_a => {}
-                Err(e) => return Err(e),
-            }
+            // Note: We don't check for the packet receipt here because another
+            // relayer may have already relayed the packet. If that's the case,
+            // we want to avoid failing the transaction and consuming
+            // unnecessary fees. Therefore, IBC core treats this as a no-op.
```

Comment:
```suggestion
            // unnecessary fees.
```
I think this statement about this case being a no-op should be removed. With it present, it becomes confusing because the comment below mentions it being not a no-op.